### PR TITLE
Add authentication, authorization, permission and wrong password alarms

### PR DESCRIPTION
**Pull request by manuelgavidia**
alarms

This is connected to dojot/dojot#222

### CODE REVIEW

File: alarms/dojot.auth.AuthenticationError.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<alarm-metamodel>
+       <domain>AuthenticationError</domain>
+       <namespace>dojot.auth</namespace>
+       <primary-subject>
+               <primary-field type="alphanumeric">module_name</primary-field>
+               <primary-field type="alphanumeric">instance_id</primary-field>
+       </primary-subject>
+       <additional-data>
+               <additional-field type="numeric">process_id</additional-field>
+               <additional-field type="numeric">signal</additional-field>
```
Comment (mmagr) [inline alarms/dojot.auth.AuthenticationError.xml]: Is this a signal like in `SIGHUP`, `SIGKILL`, and so on?

File: alarms/dojot.auth.AuthenticationError.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<alarm-metamodel>
+       <domain>AuthenticationError</domain>
+       <namespace>dojot.auth</namespace>
+       <primary-subject>
+               <primary-field type="alphanumeric">module_name</primary-field>
+               <primary-field type="alphanumeric">instance_id</primary-field>
+       </primary-subject>
+       <additional-data>
```
Comment (mmagr) [inline alarms/dojot.auth.AuthenticationError.xml]: perhaps here we could include more useful data to an admin observing the event, like
- userid 
- username
- reason (unknown user, incorrect password, DoS prevention)

File: alarms/dojot.auth.AuthorizationError.xml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<alarm-metamodel>
+       <domain>AuthorizationError</domain>
+       <namespace>dojot.auth</namespace>
+       <primary-subject>
+               <primary-field type="alphanumeric">module_name</primary-field>
+               <primary-field type="alphanumeric">instance_id</primary-field>
+       </primary-subject>
+       <additional-data>
```
Comment (mmagr) [inline alarms/dojot.auth.AuthorizationError.xml]: Perhaps here we could include more useful information to an admin observing the event, like
- userid
- resource (what is this user attempting to access)

File: alarms/dojot.auth.PermissionError.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<alarm-metamodel>
+       <domain>PermissionError</domain>
```
Comment (mmagr) [inline alarms/dojot.auth.PermissionError.xml]: What is this used for? And how is this different from `AuthorizationError`?

File: alarms/dojot.auth.WrongPassword.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<alarm-metamodel>
```
Comment (mmagr) [inline alarms/dojot.auth.WrongPassword.xml]: I think this should be merged with `AuthenticationError`

Comment (manuelgavidia) [reply alarms/dojot.auth.WrongPassword.xml]: Ok ... in the auth code both names are used, but I will remove this one.

File: alarms/dojot.auth.AuthenticationError.xml (hunk 1 above)
Comment (manuelgavidia) [reply alarms/dojot.auth.AuthenticationError.xml]: Yes, I will change with the other suggestions

File: alarms/dojot.auth.AuthenticationError.xml (hunk 2 above)
Comment (manuelgavidia) [reply alarms/dojot.auth.AuthenticationError.xml]: Ok. Will this alarm be read just by the internal components? if not, would be ok to put this details in the reason field?

File: alarms/dojot.auth.PermissionError.xml (hunk 4 above)
Comment (manuelgavidia) [reply alarms/dojot.auth.PermissionError.xml]: I willl remove this one

File: alarms/dojot.auth.AuthenticationError.xml (hunk 2 above)
Comment (mmagr) [reply alarms/dojot.auth.AuthenticationError.xml]: I'm not sure if I follow. What reason field? (the new one?)

If it is the new one, yes, I think it should be safe to let a user with this alarm's read permission to see this kind of information. (The user must be properly logged into the system, and have the right set of permissions to be able to see it)

**Pull request merged**